bottlecap/src/lifecycle/invocation/processor.rs

```diff
@@ -319,6 +329,7 @@ impl Processor {
                 _ = offsets.process_chan_tx.send(());
             }
         }
+        drop(context_buffer);
```

Comment:
are we sure we need the manual drop here? Doesn't the closure on 307 help us by 581? Is the drop necessary? If we add new code after, we have to re-lock right?

Reply:
Yeah, so since we only await at the very end, I decided to lock in the last place we modify the `context_buffer`, in reality, I'm dropping to be 100% sure that nothing strange happens. I'll add another comment to note that we drop because we await later on 332

Reply:
Instead, since I don't need to assign it to anything, I think I can refactor this by moving the first method to the bottom close to where we actually use the `get(...)` method and then use a scope to directly drop the lock 🤔